aiida_optimize/engines/_base.py

```diff
@@ -58,6 +58,13 @@ def is_finished(self) -> bool:
         Returns true if the optimization is finished.
         """
 
+    @property
+    def is_finished_ok(self) -> bool:
```

Comment:
I would like to keep the name as `is_finished_ok` to align with other AiiDA workchains like in aiida-quantumespresso. I think it should stay distinct from `is_finished` because, for example, the Convergence engine can finish without convergence, and we need to signal this to the OptimizationWorkChain in some way.

Maybe by default, it returns the value of `is_finished`, and we leave it to each engine to decide whether to implement a more rigorous check.

Reply:
Yeah makes sense. Technically it's not _exactly_ the same concept as in AiiDA (because it's not an AiiDA process), but consistency here probably still makes life easier.